packages/material-ui-unstyled/src/SwitchUnstyled/useSwitch.ts

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import {
+  unstable_useControlled as useControlled,
+  unstable_useEventCallback as useEventCallback,
+  unstable_useForkRef as useForkRef,
+  unstable_useIsFocusVisible as useIsFocusVisible,
+} from '@material-ui/utils';
+
+export interface UseSwitchProps {
+  /**
+   * If `true`, the component is checked.
+   */
+  checked?: boolean;
+  /**
+   * The default checked state. Use when the component is not controlled.
+   */
+  defaultChecked?: boolean;
+  /**
+   * If `true`, the component is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the component is read only.
+   */
+  readOnly?: boolean;
+  /**
+   * If `true`, the `input` element is required.
+   */
+  required?: boolean;
+  /**
+   * Callback fired when the state is changed.
+   *
+   * @param {object} event The event source of the callback.
+   * You can pull out the new value by accessing `event.target.value` (string).
+   * You can pull out the new checked state by accessing `event.target.checked` (boolean).
+   */
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
+  onFocus?: React.FocusEventHandler;
+  onFocusVisible?: React.FocusEventHandler;
+  onBlur?: React.FocusEventHandler;
+}
+
+export default function useSwitch(props: UseSwitchProps) {
+  const {
+    checked: checkedProp,
+    defaultChecked,
+    disabled,
+    readOnly,
+    required,
+    onChange,
+    onFocus,
+    onFocusVisible,
+    onBlur,
+  } = props;
+
+  const [checked, setCheckedState] = useControlled({
+    controlled: checkedProp,
+    default: Boolean(defaultChecked),
+    name: 'SwitchUnstyled',
```

Comment:
In the autocomplete, we make this name configurable, likely overkill for now.

Reply:
Let's add in in the future if we see it's actually needed. For now, I'm considering having `Switch` or `MuiSwitch` instead of `SwitchUnstyled`. What do you think?

Reply:
No really preferences, I personally like the name of the module, what used in the Autocomplete:
```suggestion
    name: 'useSwitch',
```

Reply:
I'm leaning towards `Switch` here. This name is visible to the developer when there's a change between uncontrolled and controlled component. Developers don't need to know that the Swich uses `useSwitch` internally, so having just `Switch` here would improve the meaning of the message IMO.